eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientInstantiationCheck.java

```diff
@@ -10,24 +10,69 @@
 import com.puppycrawl.tools.checkstyle.checks.naming.AccessModifier;
 import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
 /**
  * Verify the classes with annotation @ServiceClient should have following rules:
  * <ol>
  *   <li>No public or protected constructors</li>
  *   <li>No public static method named 'builder'</li>
  *   <li>Since these classes are supposed to be immutable, all fields in the service client classes should be final.</li>
  * </ol>
+ *
+ * All methods that has a @ServiceMethod annotation in a class annotated with @ServiceClient should follow below rules:
+ * <ol>
+ *   <li>Follows method naming pattern. Refer to Java Spec:
+ *   https://jogiles.z19.web.core.windows.net/java-spec-draft/JavaSpec.html#sec-common-service-client-patterns</li>
```

Comment:
Change link to new URL: https://azure.github.io/azure-sdk/java_introduction.html

Reply:
This doesn't look like the new URL.

Reply:
Sorry for the confusion. I clicked resolve after I resolve in my local branch. updated